app/src/main/java/at/bitfire/icsdroid/ui/TitleColorFragment.kt

```diff
@@ -4,45 +4,93 @@
 
 package at.bitfire.icsdroid.ui
 
-import android.content.Intent
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.EditText
+import androidx.core.widget.addTextChangedListener
 import androidx.fragment.app.Fragment
 import androidx.fragment.app.activityViewModels
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import at.bitfire.icsdroid.R
 import at.bitfire.icsdroid.databinding.TitleColorBinding
-import at.bitfire.icsdroid.db.LocalCalendar
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import org.joda.time.Minutes
+import org.joda.time.format.PeriodFormat
 
-class TitleColorFragment: Fragment() {
+class TitleColorFragment : Fragment() {
 
     private val model by activityViewModels<TitleColorModel>()
 
-
     override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, inState: Bundle?): View {
         val binding = TitleColorBinding.inflate(inflater, container, false)
         binding.lifecycleOwner = this
         binding.model = model
 
-        binding.color.setOnClickListener {
-            val intent = Intent(requireActivity(), ColorPickerActivity::class.java)
-            model.color.value?.let {
-                intent.putExtra(ColorPickerActivity.EXTRA_COLOR, it)
+        var firstCheck = true
```

Comment:
I will take a look at it. There should be a better way

Reply:
I haven't checked if this works, but [c233aee](https://github.com/bitfireAT/icsx5/pull/78/commits/c233aeebd1339bafa36979c51602cc0357f1d2af) should fix it. The listener is declared in a Fragment-level variable, so when we call `setOnCheckedChangeListener` multiple times, multiple listeners are not assigned (event though I'm not sure if multiple listeners are even supported).

Using this method, the first callback of `model.defaultAlarmMinutes.observe` first sets the initial value, and then assigns the listener, so that the first value update doesn't trigger it.

Take a look, I think it's fine 😉

Reply:
I have tested it and works fine 😄